dlrover/python/elastic_agent/torch/training.py

```diff
@@ -642,21 +649,17 @@ def _invoke_run(self, role: str = DEFAULT_ROLE) -> RunResult:
                 return run_result
             elif state in {WorkerState.UNHEALTHY, WorkerState.FAILED}:
                 logger.error(f"The worker fails with {run_result.failures}")
-                self._report_failure_to_master(run_result.failures)
                 self._save_ckpt_to_storage()
-                if self._remaining_failovers > 0 and not node_failed(
-                    self._log_file
-                ):
-                    logger.info(
-                        f"[{role}] Worker group {state.name}. "
-                        f"{self._remaining_failovers}/{spec.max_restarts}"
-                        f" attempts left; will restart worker group"
-                    )
-                    self._remaining_failovers -= 1
-                    self._restart_workers(self._worker_group)
-                else:
-                    self._stop_workers(self._worker_group)
-                    self._worker_group.state = WorkerState.FAILED
+
+                worker_context = WorkerContext(
```

Comment:
Need more test cases for coverage for this part involves core logic for fault tolerance.